js_modules/dagit/packages/core/src/assets/AssetEvents.tsx

```diff
@@ -139,6 +180,17 @@ export const AssetEvents: React.FC<Props> = ({
     }
   }, [loadedPartitionKeys, materializations, observations, xAxis]);
 
+  const jobRunsThatDidntMaterializeAsset = jobRunsCounts.find(
```

Comment:
I like that you've got some of the business logic down in useMostRecentRuns, think we could make it `useRecentRunWarnings(repository, assetKey, grouped[0].latest?.runId)` and put this stuff down in there as well?  It'd be kind of cool if it emitted just the two warning states.